src/daemon/Display.cpp

```diff
@@ -296,20 +296,24 @@ namespace SDDM {
         return QString();
     }
 
-    bool Display::findSessionEntry(const QDir &dir, const QString &name) const {
-        // Given an absolute path: Check that it matches dir
+    bool Display::findSessionEntry(const QStringList &dirPaths, const QString &name) const {
         const QFileInfo fileInfo(name);
-        if (fileInfo.isAbsolute() && fileInfo.absolutePath() != dir.absolutePath())
-            return false;
-
         QString fileName = name;
 
         // append extension
         const QString extension = QStringLiteral(".desktop");
         if (!fileName.endsWith(extension))
             fileName += extension;
 
-        return dir.exists(fileName);
+        for (const QDir dir: dirPaths) {
```

Comment:
Same here

Reply:
Here it would require intermediate variable, do you think it worth it?